### ISSUE TITLE
Report error on inherent associated types (for now)

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#20359]
```rust
#![feature(associated_types)]

struct Foo;
impl Foo { type Bar = int; }

fn main() {
    let _: Foo::Bar = 1;
}
```

```
<anon>:7:12: 7:20 error: ambiguous associated type; specify the type using the syntax `<Type as Foo>::Bar`
<anon>:7     let _: Foo::Bar = 1;
                    ^~~~~~~~
```

AFAICT, there's no way to use the `<... as ...>` syntax with inherent associated items (or any plans for it) so the suggestion is entirely unhelpful.

**Comment by nikomatsakis** [comment]
Inherent associated types should be reported as an error for now, they are not even remotely implemented.
